pynucastro/rates/rate.py

```diff
@@ -1687,6 +1689,71 @@ def function_string_py(self):
 
         return fstring
 
+    def function_string_cxx(self, dtype="double", specifiers="inline"):
+        """
+        Return a string containing C++ function that computes the
+        rate
+        """
+
+        for nuc in set(self.rate.reactants + self.rate.products):
+            if not nuc.partition_function and str(nuc) not in ['h1', 'n', 'he4', 'p']:
+                print(f'WARNING: {nuc} partition function is not supported by tables: set pf = 1.0 by default')
+
+        fstring = super().function_string_cxx(dtype=dtype, specifiers=specifiers, leave_open=True)
```

Comment:
This almost identical to what we have in python, with the exception of calling `super().function_string_cxx` this was very clever, to separate the partition function corrections and the evaluation of the singlesets .